tools/buildsys/src/manifest/error.rs

```diff
@@ -15,6 +15,18 @@ pub(super) enum Error {
     #[snafu(display("Cargo package graph query failed with root '{id}': {source}"))]
     CargoPackageQuerySnafu { id: PackageId, source: guppy::Error },
 
+    #[snafu(display("Cargo package is missing build-kit with name '{id}'"))]
+    CargoPackageMissingBuildKit { id: PackageId },
+
+    #[snafu(display("Cargo package is missing 'vendor' field in build-kit '{name}'"))]
+    CargoPackageMissingVendor { name: String },
+
+    #[snafu(display("Cargo package build-kit metadata is malformed '{id}': {source}"))]
+    CargoPackageMalformedBuildKit {
+        id: PackageId,
+        source: serde_json::Error,
+    },
```

Comment:
nit: at this level we should just say "package", cargo is sort of an implementation detail. Also `BuildKit` can be confusing since it's the name of one of our dependencies.

```suggestion
    #[snafu(display("Package '{id}' has no build-kit metadata"))]
    NoKitMetadata { id: PackageId },

    #[snafu(display("Package '{id}' has no 'vendor' field in build-kit metadata"))]
    NoKitVendor { id: PackageId },

    #[snafu(display("Package '{id}' has malformed build-kit metadata: {source}"))]
    MalformedKitMetadata {
        id: PackageId,
        source: serde_json::Error,
    },
```